### ISSUE TITLE
Failed to verify contract

### ISSUE THREAD

**Issue by keyboard-clacker** [rkalis/truffle-plugin-verify#70]
Hey there! I'm trying to verify a contract on mainnet and I was able to verify a linked library that we wrote and use, but when I try to verify the main contract, it fails for an unknown reason.

<details><summary>Logs...</summary>

```
λ ./node_modules/.bin/truffle run etherscan Project@<redacted> --network mainnet --debug
DEBUG logging is turned ON
Running truffle-plugin-verify v0.5.7
Verifying Project@<redacted>
Reading artifact file at /Users/kyle/workspace/project/build/contracts/Project.json
Custom address <redacted> specified
Retrieving constructor parameters from https://api.etherscan.io/api?apiKey=<redacted>&module=account&action=txlist&address=<redacted>&page=1&sort=asc&offset=1
Constructor parameters retrieved: 0x
Sending verify request with POST arguments:
{
  "apikey": "<redacted>",
  "module": "contract",
  "action": "verifysourcecode",
  "contractaddress": "<redacted>",
  "sourceCode": "<redacted>",
  "codeformat": "solidity-standard-json-input",
  "contractname": "/Users/kyle/workspace/project/contracts/Project.sol:Project",
  "compilerversion": "v0.8.2+commit.661d1103",
  "constructorArguements": ""
}
Checking status of verification request iwhcgftntajesp6q2tyq6xybrpwscd5keurpevdkfsvnsngmln
Fail - Unable to verify
Failed to verify 1 contract(s): Project@<redacted>
```

</details>

I've redacted the source code, changed the name of the contract to Project, and removed my API key and contract address, but I'm not sure if those are necessary to help debug the problem haha. Are there any steps I could take to debug this issue? One thing to note, I just deployed this contract a bit ago and it shows up on etherscan, but when I was deploying I got an error message saying "insufficient funds" on the  `Saving migration to chain.` step in Truffle. Also there is a linked library as well which I was able to successfully verify, but it appears as if in my `Project` contract, the `libraries` key is an empty object, even when trying to verify them at the same time. The only thing I can think of is that that library that we _were_ able to verify doesn't have any imports, whereas in `Project` we do import other contracts.

Also for what it's worth, I was able to deploy and verify the exact same contract(s) on rinkeby right after I got the error, just to test to see if maybe that was broken as well.

Thanks in advance for any help 🍻 
Kyle

**Comment by keyboard-clacker** [comment]
Update on this, I just realized it may be because my truffle migration failed at the "Saving migration to chain" step. Once I deployed and had that step pass by decreasing my gas limit (apparently truffle uses your default global gas limit for saving to chain, which seems very high), I was able to redeploy and verify the second contract that initially was failing verification.

Think maybe it's worth adding that to the README as a root cause? Happy to open a PR
